src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -575,20 +553,30 @@ const DateTimePicker = ({
   const onApplyClick = () => {
     setIsExpanded(false);
     const value = renderValue();
+    const returnValue = {
+      timeRangeKind: value.kind,
+      timeRangeValue: null,
+    };
     switch (value.kind) {
       case PICKER_KINDS.ABSOLUTE:
+        returnValue.timeRangeValue = value.absolute;
         setLastAppliedValue(value.absolute);
         break;
       case PICKER_KINDS.RELATIVE:
+        returnValue.timeRangeValue = value.relative;
         setLastAppliedValue(value.relative);
         break;
       default:
+        returnValue.timeRangeValue = value.preset;
         setLastAppliedValue(value.preset);
         break;
     }
 
     if (onApply) {
-      onApply(value);
+      if (__DEV__) {
+        console.log('onApply callback return value', { returnValue }); // eslint-disable-line no-console
+      }
```

Comment:
I don't think this is necessary. A breakpoint can be placed here in the dev debugger to view the return object.
```suggestion
```